solidity/random-beacon/contracts/RandomBeacon.sol

```diff
@@ -778,13 +910,22 @@ contract RandomBeacon is IRandomBeacon, Ownable {
             "Invalid group members"
         );
 
-        uint256 slashingAmount = relay.submitEntry(entry, group.groupPubKey);
+        uint96 slashingAmount = relay.submitEntry(entry, group.groupPubKey);
 
         if (slashingAmount > 0) {
             address[] memory groupMembersAddresses = sortitionPool
                 .getIDOperators(groupMembers);
 
-            try staking.slash(slashingAmount, groupMembersAddresses) {
+            address[] memory stakingProvidersAddresses = new address[](
```

Comment:
I tried to extract this piece of code to the `Authorizations` library (code snippet below) to reduce transaction costs, but unfortunately it ended up being even more expensive:
```
        method            |   min   |   max   |   avg
-------------------------------------------------------
submitRelayEntry          |  213867 | 3735913 | 1779221 
reportRelayEntryTimeout   | 3683609 | 3701610 | 3696162
reportUnauthorizedSigning |    -    |    -    | 3816486
```

```solidity
    function operatorsToStakingProviders(
        Data storage self,
        address[] memory operators
    ) external returns (address[] memory) {
        address[] memory stakingProviders = new address[](operators.length);
        for (uint256 i = 0; i < operators.length; i++) {
            stakingProviders[i] = self.operatorToStakingProvider[operators[i]];
        }

        return stakingProviders;
    }
```

Reply:
We will need to do some diligence here and figure out where does the cost of gas come from. If it is real `TokenStaking` or something else. `TokenStaking` writes to the queue so the cost could increase by 1.2M for a whole group slashed but what we see is an increase by 3.28M for `reportUnauthorizedSigning`.

Reply:
For a group of 100, the total cost of `reportUnauthorizedSigning` would be `5 726 857`. It turns out that: 
- `sortitionPool.getIDOperators` costs `326 864`
- `operatorToStakingProvider` loop costs `249 402`
- `tokenStaking.seize` costs `4 858 014`

Reply:
From the `TokenStaking` perspective it's `slashingQueue.push` that produces such a cost burden. For `tokenStaking.seize` call mentioned above the `slashingQueue.push` execution consumes `4 546 619` out of the `4 858 014`.

Reply:
Addressed in https://github.com/threshold-network/solidity-contracts/pull/87.